docs/user-guide/src/capm3/pivoting.md

```diff
@@ -0,0 +1,172 @@
+
+## Prerequisite
+
+* **It is mandatory to use `clusterctl` for both the bootstrap and target cluster.**
+
+  If the provider components are not installed using `clusterctl`, it will not
+  be able to identify the objects to move.  Initializing the cluster using
+  `clusterctl` essentially adds the following labels in the CRDs of each related object.
+
+  ```yaml
+  labels:
+  - clusterctl.cluster.x-k8s.io: ""
+  - cluster.x-k8s.io/provider: "<provider-name>"
+  ```
+
+  So if the clusters are not initialized using `clusterctl`, all the CRDS of the
+  objects to be moved to target cluster needs to have these labels both in
+  bootstrap cluster and target cluster before performing the move.
+
+  **Note**: This is not recommended, since
+  the way `clusterctl` identifies objects to manage might change in the future, so
+  it's always safe to install CRDs and controllers through the `clusterctl init` sub-command.
+
+* **BareMetalHost objects have correct status annotation.**
+
+  Since BareMetalHost (BMH) _status_ holds important information regarding the BMH itself, BMH with
+  status has to be moved and it has to be reconstructed with correct status in
+  target  cluster before it is being reconciled. This is now done through BMH
+  status annotation in [BMO](../bmo/introduction.md).
+
+* **Maintain connectivity towards provisioning network.**
+
+  Baremetal machines boot over a network with a DHCP server. This requires
+  maintaining a fixed IP end points towards the provisioning network. This is
+  achieved through _keepalived_. A new container is added namely
+  _ironic-endpoint-keepalived_ in the ironic deployment which maintains the Ironic Endpoint using
+  keepalived. The motivation behind maintaining Ironic Endpoint with Keepalived
+  is to ensure that the Ironic Endpoint IP is also passed onto the target
+  cluster control plane. This also guarantees that once moving is done and the
+  management cluster is taken down, target cluster controlplane can re-claim
+  the Ironic endpoint IP through keepalived. The end goal is to make Ironic
+  endpoint reachable in the target cluster.
+
+* **BMO is deployed as part of CAPM3.**
+
+  If not, it has to be deployed before the `clusterctl init` and the BMH CRDs
+  need to be labeled accordingly manually. Separate labeling for BMH CRDs is required
+  because since CAPM3 release [v0.5.0](https://github.com/metal3-io/cluster-api-provider-metal3/releases/tag/v0.5.0)
+  BMO/BMH CRDs are not deplopyed as part of CAPM3 deployment anymore.
+  This is a prerequisite for both the management and the target cluster.
+
+* **Objects should have a proper owner reference chain.**
+
+  `clusterctl move` moves all the objects to the target cluster following the
+[owner reference chain](https://cluster-api.sigs.k8s.io/clusterctl/provider-contract.html#ownerreferences-chain).
+So, it is necessary to verify that all the desired objects that needs to
+be moved to the target cluster have a proper owner reference chain.
+
```

Comment:
```suggestion
1. **It is mandatory to use `clusterctl` for both the bootstrap and target cluster**

      If the provider components are not installed using `clusterctl`, it will not
      be able to identify the objects to move.  Initializing the cluster using
      `clusterctl` essentially adds the following labels in the CRDs of each related object.

      ```yaml
      labels:
      - clusterctl.cluster.x-k8s.io: ""
      - cluster.x-k8s.io/provider: "<provider-name>"
      ```

      So if the clusters are not initialized using `clusterctl`, all the CRDS of the
      objects to be moved to target cluster needs to have these labels both in
      bootstrap cluster and target cluster before performing the move.

      **Note**: This is not recommended, since
      the way `clusterctl` identifies objects to manage might change in the future, so
      it's always safe to install CRDs and controllers through the `clusterctl init` sub-command.

1. **BareMetalHost objects have correct status annotation**

   Since BareMetalHost (BMH) _status_ holds important information regarding the BMH itself, BMH with
   status has to be moved and it has to be reconstructed with correct status in
   target  cluster before it is being reconciled. This is now done through BMH
   status annotation in [BMO](../bmo/introduction.md).

1. **Maintain connectivity towards provisioning network**

   Baremetal machines boot over a network with a DHCP server. This requires
   maintaining a fixed IP end points towards the provisioning network. This is
   achieved through _keepalived_. A new container is added namely
   _ironic-endpoint-keepalived_ in the ironic deployment which maintains the Ironic Endpoint using
   keepalived. The motivation behind maintaining Ironic Endpoint with Keepalived
   is to ensure that the Ironic Endpoint IP is also passed onto the target
   cluster control plane. This also guarantees that once moving is done and the
   management cluster is taken down, target cluster controlplane can re-claim
   the Ironic endpoint IP through keepalived. The end goal is to make Ironic
   endpoint reachable in the target cluster.

1. **BMO is deployed as part of CAPM3**

   If not, it has to be deployed before the `clusterctl init` and the BMH CRDs
   need to be labeled accordingly manually. Separate labeling for BMH CRDs is required
   because since CAPM3 release [v0.5.0](https://github.com/metal3-io/cluster-api-provider-metal3/releases/tag/v0.5.0)
   BMO/BMH CRDs are not deplopyed as part of CAPM3 deployment anymore.
   This is a prerequisite for both the management and the target cluster.

1. **Objects should have a proper owner reference chain**

   `clusterctl move` moves all the objects to the target cluster following the
   [owner reference chain](https://cluster-api.sigs.k8s.io/clusterctl/provider-contract.html#ownerreferences-chain).
   So, it is necessary to verify that all the desired objects that needs to
   be moved to the target cluster have a proper owner reference chain.
```